### DISCUSSION TITLE
Update is giving a git error

### DISCUSSION THREAD

**Discussion by bilenemek** [firedrakeproject/firedrake#3045] [Firedrake support]
Hi All

A simple firedrake-update on a working firedrake installation gives a git error.

Updating the git repository for firedrake
Traceback (most recent call last):
  File "/home/emek/firedrake/bin/firedrake-update", line 1840, in <module>
    git_update("firedrake")
  File "/home/emek/firedrake/bin/firedrake-update", line 891, in git_update
    check_call(["git", "pull", "--recurse-submodules"])
  File "/home/emek/firedrake/bin/firedrake-update", line 592, in check_call
    log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=os.environ).decode())
  File "/usr/lib/python3.8/subprocess.py", line 415, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'pull', '--recurse-submodules']' returned non-zero exit status 1.

I want to install netgen and opencascade. So I add to configuration.json "netgen": true but this error is not related to this issue. Is there a change in git repository or so that it cannot find?

I am running it from the upper level (virtual environment folder) after acivating venv. How to fix it?

Best, Emek

**Comment by ksagiyam** [comment]
Could you please upload `$VIRTUAL_ENV/firedrake-update.log?`

**Comment by bilenemek** [comment]
2023-07-27 08:26:17,213 INFO   BLAS config is {}
2023-07-27 08:26:17,214 INFO   Configuration saved to /home/emek/firedrake/.configuration.json
2023-07-27 08:26:17,214 INFO   Updating the git repository for firedrake
2023-07-27 08:26:17,214 DEBUG  Old path '/home/emek/firedrake/src'
2023-07-27 08:26:17,214 DEBUG  Pushing path '/home/emek/firedrake/src/firedrake'
2023-07-27 08:26:17,214 DEBUG  Running command 'git rev-parse HEAD'
2023-07-27 08:26:17,219 DEBUG  Running command 'git pull --recurse-submodules'
2023-07-27 08:26:18,005 DEBUG  error: Your local changes to the following files would be overwritten by merge:
	.github/workflows/build.yml
	.github/workflows/docs.yml
	.github/workflows/lint.yml
	.github/workflows/zenodo-canary.yml
	AUTHORS
	CITATION.md
	LICENSE
	Makefile
.
.
.
Aborting
Updating d16e15ee7..0ec02b2d8

2023-07-27 08:26:18,005 DEBUG  Popping path '/home/emek/firedrake/src/firedrake'
2023-07-27 08:26:18,005 DEBUG  New path '/home/emek/firedrake/src'
2023-07-27 08:26:18,008 INFO   

Update log saved in /home/emek/firedrake/firedrake-update.log

**Comment by bilenemek** [comment]
Thanks for the answer. I guess that the issue is the wrong path but I cannot find where it is set. It is not related where I run the command. After enabling venv, since firedrak-update is in bin, it is possible to call it in different directories. I tried also different ones but the error is the same.

**Comment by ksagiyam** [comment]
Actually, you do not want to change `configuration.json` manually.
If you want to add netgen, `firedrake-update --netgen` should work.

**Comment by bilenemek** [comment]
It was my first trial, but it ends up with

Running /home/emek/firedrake/bin/firedrake-update --netgen
usage: firedrake-update [-h] [--no-update-script] [--rebuild] [--rebuild-script] [--slepc] [--opencascade] [--tinyasm]
                        [--honour-petsc-dir] [--petsc-int-type {int32,int64}] [--honour-pythonpath] [--clean] [--verbose]
                        [--install PACKAGES] [--pip-install PIP_PACKAGES] [--install-help] [--cache-dir CACHE_DIR]
                        [--complex | --real] [--documentation-dependencies] [--show-dependencies] [--remove-build-files]
                        [--with-blas WITH_BLAS]
firedrake-update: error: unrecognized arguments: --netgen

Therefore, I wanted to add it to the json file. Now I deleted, but still the problem is not this one. It cannot pass the git pull because of wrong path. I am using Ubuntu 20.04 and the installation is fine, firedrake is running very nice in parallel for some time.

**Comment by ksagiyam** [comment]
I see. That is a separate issue. Could you post the output of `firedrake-status`?

**Reply by bilenemek** [reply]
Sorry, I could not make it look better than the output below.

**Comment by bilenemek** [comment]
```
Firedrake Configuration:
    package_manager: True
    minimal_petsc: False
    mpicc: None
    mpicxx: None
    mpif90: None
    mpiexec: None
    disable_ssh: False
    honour_petsc_dir: False
    with_parmetis: False
    slepc: False
    packages: []
    honour_pythonpath: False
    opencascade: True
    tinyasm: False
    petsc_int_type: int32
    cache_dir: /home/emek/firedrake/.cache
    complex: False
    remove_build_files: False
    with_blas: None
Additions:
    None
Environment:
    PYTHONPATH: None
    PETSC_ARCH: None
    PETSC_DIR: None
Status of components:
---------------------------------------------------------------------------
|Package             |Branch                        |Revision  |Modified  |
---------------------------------------------------------------------------
|COFFEE              |master                        |70c1e66   |True      |
|FInAT               |master                        |4112fb8   |True      |
|PyOP2               |master                        |a0259a91  |True      |
|fiat                |master                        |629fe77   |True      |
|firedrake           |master                        |d16e15ee7 |True      |
|h5py                |firedrake                     |78531f08  |True      |
|libspatialindex     |master                        |4768bf3   |True      |
|libsupermesh        |master                        |69012e5   |True      |
|loopy               |main                          |3988272b  |True      |
|petsc               |firedrake                     |729b7b4f1b|True      |
|pyadjoint           |master                        |0dcfe22   |True      |
|tsfc                |master                        |351994d   |True      |
|ufl                 |master                        |9135be16  |True      |
---------------------------------------------------------------------------
```

**Comment by ksagiyam** [comment]
I do not know why all the components are claimed to be modified, and I think this is why `firedrake-update` is failing.
Have you modified these packages?
You could try to directly figure out why this happend, but, reinstalling from scratch might be easier.

`firedrake-status` normally looks like this:
```
...
---------------------------------------------------------------------------
|Package             |Branch                        |Revision  |Modified  |
---------------------------------------------------------------------------
|COFFEE              |master                        |70c1e66   |False     |
|FInAT               |master                        |47f6c37   |False     |
|PyOP2               |master                        |d230953b  |False     |
|fiat                |master                        |8c66270   |False     |
|firedrake           |master                        |0ec02b2d8 |False     |
|h5py                |firedrake                     |6cc4c912  |False     |
|icepack             |master                        |0c17259   |False     |
|libspatialindex     |master                        |4768bf3   |True      |
|libsupermesh        |master                        |b145b65   |False     |
|loopy               |main                          |8158afdb  |False     |
|petsc               |firedrake                     |9364cb008b2|False     |
|pyadjoint           |master                        |0378c81   |False     |
|pytest-mpi          |main                          |a478bc8   |False     |
|slepc               |firedrake                     |e438e4993 |False     |
|tsfc                |master                        |6f72c9c   |False     |
|ufl                 |master                        |3c62318c  |False     |
---------------------------------------------------------------------------
```
